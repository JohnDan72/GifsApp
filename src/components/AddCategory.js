import React, { useState } from 'react';
import { Input, InputGroup } from 'rsuite';

const AddCategory = () => {
    const [inputValue, setInputValue] = useState('Hola mundo')
    return (
        <>
            <InputGroup size="md">
                <Input placeholder="nueva categorÃ­a" />
                {/*onClick={(e) => handleAdd()}*/}
                <InputGroup.Button >
                    Agregar
                </InputGroup.Button>

            </InputGroup>
        </>
    );
}

export default AddCategory;