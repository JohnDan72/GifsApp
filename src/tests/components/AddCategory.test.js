import { shallow } from "enzyme"
import React from "react";
import AddCategory from "../../components/AddCategory"



describe('Pruebas de AddCategoryComponent' , () => {
    const [ categories , setCategories ] = [['cat 1','cat 2','cat 3'], jest.fn()];
    let wrapper = shallow( <AddCategory categories={categories} setCategories={setCategories} /> );

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory categories={categories} setCategories={setCategories} /> );
    });

    test('should render a AppCategory component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should change input value of input', () => {
        const input = wrapper.find('Input');
        input.simulate('change','matrix');
        // console.log(input.props());
        // console.log(wrapper.find('Input').props());
        expect(wrapper.find('Input').prop('value')).not.toBe('');
    });

    test('No debe postear el form', () => {
        wrapper.find('Form').simulate('submit',true); //se manda el argunmento (como no lo uso le envio un true)
        
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Debe postear el form y limpiar caja de texto', () => {
        const input = wrapper.find('Input')
        input.simulate('change','matrix');
        // console.log(input.props());
        wrapper.find('Form').simulate('submit',true); //se manda el argunmento (como no lo uso le envio un true)
        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('Input').prop('value')).toBe('');
    });


    
})