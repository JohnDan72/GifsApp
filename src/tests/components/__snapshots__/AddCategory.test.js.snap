// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de AddCategoryComponent should render a AppCategory component 1`] = `
<Fragment>
  <Form
    onSubmit={[Function]}
  >
    <InputGroup
      size="md"
    >
      <Input
        onChange={[Function]}
        placeholder="nueva categorÃ­a"
        value=""
      />
      <InputGroupButton
        type="submit"
      >
        Agregar
      </InputGroupButton>
    </InputGroup>
  </Form>
</Fragment>
`;
